### PR TITLE
change _Pragma to #pragma

### PR DESCRIPTION
**Pull request by yajiedesign**
## Description ##
change _Pragma to #pragma
_Pragma not work with msvc

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)

### CODE REVIEW

**Comment by yajiedesign** [comment]
https://github.com/apache/incubator-mxnet/issues/18374

**Comment by zhreshold** [comment]
`#pragma` won't work in a macro, what's the version of msvc? At least 2019 supports `_Pragma`

File: src/io/batchify.cc [hunk 1]
```diff
@@ -150,7 +150,7 @@ class StackBatchify : public BatchifyFunction {
         }
         int sbs = static_cast<int>(bs);
         MSHADOW_TYPE_SWITCH_WITH_BOOL(dtype, DType, {
-          _Pragma("omp parallel for num_threads(bs)")
+          #pragma omp parallel for num_threads(bs)
```
Comment (zhreshold) [inline src/io/batchify.cc:153]: use `__pragma` for msvc if it does not support `_Pragma`

Comment (wkcn) [reply src/io/batchify.cc:153]: We can unify the macro like
```c++
#ifndef _MSC_VER
#define __pragma(id) _Pragma(#id)
#endif
```

**Comment by yajiedesign** [comment]
I'll try upgrading vs

**Comment by yajiedesign** [comment]
Cannot be resolved by upgrade vs.

**Comment by yajiedesign** [comment]
@mxnet-bot run ci [windows-cpu]

File: src/io/batchify.cc [hunk 2]
```diff
@@ -37,6 +37,17 @@
 
 namespace mxnet {
 namespace io {
+
+#ifdef _MSC_VER
+  #if _MSC_VER < 1925
+    #define omp_parallel __pragma(omp parallel for num_threads(bs))
```
Comment (zhreshold) [inline src/io/batchify.cc:43]: Can you make `bs` argument of the macro?

**Comment by yajiedesign** [comment]
@leezu Because CUDA. The old version of CUDA does not support the new version of msvc.
CI is only one CUDA version. but CD requires multiple versions of CUDA. cannot upgrade MSVC.

**Comment by leezu** [comment]
Thanks for clarifying. In that case, should we either drop the old Cuda versions for MXNet 2 or switch the CI to an older MSVC version? What do you think?

**Comment by leezu** [comment]
(Soon Cuda 11 will be released and we'll drop Cuda 9 support in any case)

**Comment by yajiedesign** [comment]
Keep the MSVC of CI compatible with the minimum supported CUDA version.
Then upgrade MSVC when drop CUDA version support.

**Comment by leezu** [comment]
MXNet 2 will be released after Cuda 11 release, so there is no Cuda 9 support in MXNet 2 currently

**Comment by leezu** [comment]
Yes, that's an option. I'm not convinced though that only fixing this current incompatibility without introducing any mechanism of preventing future problems is helpful. 
So if you think we need to support MSVC 2017, then let's also turn it on in the CI. If not, then let's just use MSVC 2019 on the CD. But supporting MSVC 2017 without having any CI checks would just cause unnecessary problems down the road (with the CD breaking often)

**Comment by yajiedesign** [comment]
@mxnet-bot run ci [macosx-x86_64]

**Comment by yajiedesign** [comment]
@zhreshold

**Comment by leezu** [comment]
@yajiedesign please clarify how you plan to address the CI CD discrepancy?

**Pull request merged**